src/extra/utils/ftd.pyx

```diff
@@ -0,0 +1,517 @@
+recording of a continguous analog signal.
+"""
+
+
+from cython cimport floating
+from cython.view cimport contiguous
+from libc.math cimport M_PI, sin, fabs, floor, ceil, round
+from libc.stdlib cimport malloc, free
+
+import numpy as np
+
+
+ctypedef floating data_t
+ctypedef floating[::contiguous] data_array_t
+
+cpdef enum EdgeInterpolation:
+    # Interpolation method to find edge position.
+
+    NEAREST = 0
+    LINEAR = 1
+    SPLINE = 2
+    SINC = 3
+
+"""Sinc interpolation parameters."""
+cdef int sinc_window = 200
+cdef int sinc_search_iterations = 10
+
+
+def config_sinc_interpolation(window=None, search_iterations=None):
+    """Configure sinc interpolation.
+
+    As a consequence of the Nyquist theorem, sinc interpolation allows
+    to reconstruct a continuous-time function $x(t)$ up to a certain
+    bandwidth $1/T$ from a sequence of real numbers $x[n]$:
+
+    $$
+    x(t) = \sum_{n=-\infty}^{\infty} x[n] \, {\rm sinc}\left(\frac{t - nT}{T}\right)
+    $$
+
+    This can be used to find the optimal edge position between samples
+    with the fast timing discriminators in this module. In particular
+    for fast slopes consisting of only a few sample points, this will
+    generally yield significantly better results than linear
+    interpolation, which tends to shift points towards the middle. For
+    the constant fraction discriminator specifically, this can also
+    be used to enable the use of real delay values.
+
+    This method has generally a large performance impact, but can be
+    somewhat tuned by parameters:
+
+    * `window` specifies the number of samples before and after the
+        interpolation points actually used to evaluate $x(t), i.e. the
+        finite boundaries to approximate the infinite sum above. By
+        default, up to 200 samples in each direction are used.
+
+    * `search_iterations` specifies the number of binary search steps
+        taken to find the optimal edge position from interpolated values
+        in between two samples. The maximal resolution in samples the
+        interpolation can therefore achieve is $2^-{\rm search_iterations}$.
+
+    When set, these parameters apply to all discriminator
+    implementations and all their use of sinc interpolation.
+
+    Args:
+        window (int, optional): Sample window used around the
+            interpolated point, unchanged if omitted.
+        search_iterations (int, optional): Number of iterations used in
+            binary search for closest function argument, unchanged if
+            omitted.
+
+    Returns
+        (window, search_iterations) Tuple of current values.
+    """
+
+    global sinc_window, sinc_search_iterations
+
+    if window is not None:
+        sinc_window = <int>window
+
+    if search_iterations is not None:
+        sinc_search_iterations = <int>search_iterations
+
+    return dict(window=sinc_window, search_iterations=sinc_search_iterations)
+
+
+cdef inline double _sinc(double x) nogil:
```

Comment:
```suggestion
cdef inline double _sinc(double x) noexcept nogil:
```

From Cython 3.0, functions like this one are considered able to raise exceptions by default, so after each call it will insert a check for an exception.

https://docs.cython.org/en/latest/src/userguide/migrating_to_cy30.html#exception-values-and-noexcept

The same applies to `_sinc_interp`, `_cfd_sinc_interp` and `_dled_sinc_interp`, I think. Naturally if we're using `noexcept` we should be confident these can't raise exceptions.

Reply:
TIL, thanks!

Due to a change below, I did not add it to `_cfd_sinc_interp`.